pkg/services/ngalert/state/manager.go

```diff
@@ -128,20 +128,21 @@ func (st *Manager) Run(ctx context.Context) error {
 	return nil
 }
 
-func (st *Manager) Warm(ctx context.Context, rulesReader RuleReader, instanceReader InstanceReader) {
+func (st *Manager) Warm(ctx context.Context, orgReader OrgReader, rulesReader RuleReader, instanceReader InstanceReader) {
 	logger := st.log.FromContext(ctx)
 
-	if st.instanceStore == nil {
-		logger.Info("Skip warming the state because instance store is not configured")
+	if orgReader == nil || rulesReader == nil || instanceReader == nil {
+		logger.Error("Unable to warm state cache, missing required store readers")
 		return
 	}
 
 	startTime := time.Now()
 	logger.Info("Warming state cache for startup")
 
-	orgIds, err := instanceReader.FetchOrgIds(ctx)
+	orgIds, err := orgReader.FetchOrgIds(ctx)
```

Comment:
The only downside is that previously, we only retrieved organization IDs with alert rules and a state. Now, we will fetch all organization IDs, try to list their alert rules and retrieve their state, even if the organization has none. However, this is fine during startup.